airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -386,6 +386,18 @@ public List<Job> listJobsWithStatus(final ConfigType configType, final JobStatus
     return listJobsWithStatus(Sets.newHashSet(configType), status);
   }
 
+  @Override
+  public List<Job> listJobsForConnectionWithStatuses(final UUID connectionId, final Set<ConfigType> configTypes, final Set<JobStatus> statuses)
+      throws IOException {
+    return jobDatabase.query(ctx -> getJobsFromResult(ctx
+        .fetch(BASE_JOB_SELECT_AND_JOIN + "WHERE " +
+            "scope = ? AND " +
+            "CAST(config_type AS VARCHAR) IN " + Sqls.toSqlInFragment(configTypes) + " AND " +
```

Comment:
Do we need the Cast here since we are doing a custom query.

Reply:
I just tried removing the cast and the test still passed, so it looks like this was not needed after all. Good callout